plugins/pipelines/lib/samson_pipelines/samson_plugin.rb

```diff
@@ -0,0 +1,40 @@
+module SamsonPipelines
+  class Engine < Rails::Engine
+  end
+
+  class << self
+    def start_pipelined_stages(job, success, output)
+      return if !success || !job.deploy || job.deploy.stage.next_stage_ids.empty?
+
+      deploy_service = DeployService.new(job.deploy.user)
+      Stage.find(job.deploy.stage.next_stage_ids).each do |next_stage|
+        start_next_stage(next_stage, job, deploy_service, output)
+      end
+    rescue => ex
+      output.write("Failed to start the pipelined stages: #{ex.message}")
+    end
+
+    def start_next_stage(next_stage, current_job, deploy_service, output)
+      new_deploy = deploy_service.deploy!(next_stage, reference: current_job.deploy.commit)
```

Comment:
@grosser 
> could we pass buddy: current_job.deploy.buddy and save the elsif below ?

That would require changing the deploy_service code, so leaving it like this for the moment. Also the elsif would have to stay anyway unless we changed deploy_service further to auto-confirm deploy if buddy is set.
I'll leave that for another PR.